Example/Tests/BONHumanReadableStringTestCase.m

```diff
@@ -0,0 +1,208 @@
+
++ (UIImage *)dummyImageOfSize:(CGSize)size scale:(CGFloat)scale
+{
+    UIGraphicsBeginImageContextWithOptions(size, YES, scale);
+
+    [[UIColor redColor] setFill];
+    UIRectFill(CGRectMake(0.0, 0.0, size.width, size.height));
+
+    UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    return image;
+}
+
+- (void)testDoubleRoundedString
+{
+    NSString *basic = BONDoubleRoundedString(1.0);
+    XCTAssertEqualObjects(basic, @"1");
+
+    NSString *epsilonTest = BONDoubleRoundedString(1.0000001);
+    XCTAssertEqualObjects(epsilonTest, @"1");
+
+    NSString *negative = BONDoubleRoundedString(-1.0);
+    XCTAssertEqualObjects(negative, @"-1");
+
+    NSString *smallFractional = BONDoubleRoundedString(7 + kBONTwoThirds);
+    XCTAssertEqualObjects(smallFractional, @"7.667");
+
+    NSString *largeFractional = BONDoubleRoundedString(13682 + kBONOneThird);
+    XCTAssertEqualObjects(largeFractional, @"13682.333");
+
+    NSString *halfFractional = BONDoubleRoundedString(12345.5);
+    XCTAssertEqualObjects(halfFractional, @"12345.5");
+}
+
+- (void)testPrettyCGSize
+{
+    NSString *basic = BONPrettyStringFromCGSize(CGSizeMake(1.0, 1.0));
+    XCTAssertEqualObjects(basic, @"1x1");
+
+    NSString *fractions = BONPrettyStringFromCGSize(CGSizeMake(13.5, 827 + kBONOneThird));
+    XCTAssertEqualObjects(fractions, @"13.5x827.333");
+
+    NSString *bigFractions = BONPrettyStringFromCGSize(CGSizeMake(2468.0 + kBONTwoThirds, -4272.5));
+    XCTAssertEqualObjects(bigFractions, @"2468.667x-4272.5");
+}
+
+- (void)testDummyImageMaker
+{
+    UIImage *image1x = [self.class dummyImageOfSize:CGSizeMake(1.0, 1.0) scale:1.0];
+    XCTAssertEqualWithAccuracy(image1x.size.width, 1.0, kBONEpsilon);
+    XCTAssertEqualWithAccuracy(image1x.size.height, 1.0, kBONEpsilon);
+
+    UIImage *image2x = [self.class dummyImageOfSize:CGSizeMake(1.0, 1.0) scale:2.0];
+    XCTAssertEqualWithAccuracy(image2x.size.width, 1.0, kBONEpsilon);
+    XCTAssertEqualWithAccuracy(image2x.size.height, 1.0, kBONEpsilon);
+
+    UIImage *image3x = [self.class dummyImageOfSize:CGSizeMake(1.0, 1.0) scale:2.0];
+    XCTAssertEqualWithAccuracy(image3x.size.width, 1.0, kBONEpsilon);
+    XCTAssertEqualWithAccuracy(image3x.size.height, 1.0, kBONEpsilon);
+
+    UIImage *imageFractional3x = [self.class dummyImageOfSize:CGSizeMake(3.0 + kBONOneThird, 10.0 + kBONTwoThirds) scale:3.0];
+    XCTAssertEqualWithAccuracy(imageFractional3x.size.width, 3.333333333, kBONEpsilon);
+    XCTAssertEqualWithAccuracy(imageFractional3x.size.height, 10.66666666, kBONEpsilon);
+}
+
+- (void)testBasicImageReplacementString
+{
+    UIImage *image = [self.class dummyImageOfSize:CGSizeMake(36.0, 36.0) scale:1.0];
+    BONChain *imageChain = BONChain.new.image(image);
+    BONChain *textChain = BONChain.new.string(@"concatenate me!");
+
+    [imageChain appendLink:textChain];
+
+    BONAssertEquivalentStrings(imageChain.attributedString, @"{image36x36}concatenate me!");
```

Comment:
@adamhrz remember the issues we had, writing unit tests for complex attributed strings? This should help. What do you think about the format of `{image36x36}`? There's also a variant where you can omit the image size, if you need determinism: `{image}`. Do you think that format looks OK?